projects/admin/src/app/circulation/main-checkin-checkout/main-checkin-checkout.component.ts

```diff
@@ -135,39 +135,56 @@ export class MainCheckinCheckoutComponent implements OnInit, NoPendingChange {
   }
 
   getItem(barcode: string) {
-    const item = this.items.find(currItem => currItem.barcode === barcode);
-    if (item) {
-      if (this.patron) {
-        item.currentAction = ItemAction.checkin;
-        this.searchText = '';
-      } else {
-        this.toastService.warning(this.translate.instant('The item is already in the list.'), this.translate.instant('checkin'));
-      }
-    } else {
-      this.itemsService.getItem(barcode, this.patron.pid).subscribe(
-        (newItem) => {
-          if (newItem === null) {
-            this.toastService.warning(this.translate.instant('item not found!'), this.translate.instant('checkin'));
+    this.itemsService.getItem(barcode, this.patron.pid).subscribe(
+      (item) => {
+        // check if operation is allowed on this item for current logged user
+        if (this.loggedUser.library.organisation.pid !== item.organisation.pid) {
+          this.toastService.warning(
+            this.translate.instant('No action allowed: the item belongs to another organisation.'),
+            this.translate.instant('checkin')
+          );
+          return;
+        }
+        // check if item is already in the user items list. If yes, this is a check-in
+        const itemInList = this.items.find(currItem => currItem.barcode === barcode);
```

Comment:
probably you can use some instead of find:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some

Reply:
done